### PR TITLE
Feature - 310

### PR DESCRIPTION
**Pull request by quentin41500**
Here is an attempt to remove the usage of calendar in favor of LocalDate from java.time. LocalDate api is just much better, and this also takes care of a lot of the issues we were facing with timezone and others. Ultimately, LocalDate is just the same thing as our own CalendarDay, so we would remove CalendarDay completely.

I'm currently looking for feedback and comments on how we could make this better, as well as fixing the potential bugs. If people are interested, make sure to 👍so we can push to make it in.

TODO:
- Right now, we have to initiate AndroidThreeTen `AndroidThreeTen.init(this);` in the application class. What are the alternatives?

# Fixes
- Removed java.util.Calendar in favor of java.time.LocalDate (ThreeTenABP)
- Timezone issues
- Month format starting at 1 to 12 instead of 0 to 11.
- General cleanup
- Fixes https://github.com/prolificinteractive/material-calendarview/issues/447
- Fixes https://github.com/prolificinteractive/material-calendarview/issues/805

### CODE REVIEW

**Comment by tevjef** [comment]
@quentin41500 Though I generally dislike this technique, you can try initializing the AndroidThreeTen from a [content provider](https://medium.com/@andretietz/auto-initialize-your-android-library-2349daf06920).

File: dependencies.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ext {
-    minSdkVersion = 14
+    minSdkVersion = 16
```
Comment (tevjef) [inline dependencies.gradle]: Why did this change?

Comment (quentin41500) [reply dependencies.gradle]: `minSdkVersion` has to 15 for 310abp so i figured might as well update it to the latest supported

File: library/build.gradle [hunk 2]
```diff
@@ -24,6 +24,7 @@ android {
 }
 
 dependencies {
+    implementation rootProject.ext.threeTenAbp
```
Comment (quentin41500) [inline library/build.gradle]: Im curious if we should leak this through our api.

File: library/src/main/java/com/prolificinteractive/materialcalendarview/CalendarDay.java [hunk 3]
```diff
@@ -5,114 +5,67 @@
...
-     * Get a new instance set to today
-     *
-     * @return CalendarDay set to today's date
+     * Cache for calls to {@linkplain #getDate()}
      */
-    @NonNull
-    public static CalendarDay today() {
-        return from(CalendarUtils.getInstance());
-    }
+    @NonNull private final transient LocalDate _calendar;
```
Comment (tevjef) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/CalendarDay.java]: No need for this to be transient.

File: library/build.gradle (hunk 2 above)
Comment (tevjef) [reply library/build.gradle]: Absolutely not.

File: library/src/main/java/com/prolificinteractive/materialcalendarview/CalendarDay.java [hunk 4]
```diff
@@ -121,16 +74,16 @@ public CalendarDay(int year, int month, int day) {
...
     /**
-     * Get the month, represented by values from {@linkplain Calendar}
+     * Get the month, represented by values from {@linkplain LocalDate}
      *
-     * @return the month of the year as defined by {@linkplain Calendar}
+     * @return the month of the year as defined by {@linkplain LocalDate}
      */
     public int getMonth() {
-        return month;
+        return _calendar.getMonthValue();
```
[the remark this answers is not in the document]
Comment (quentin41500) [reply library/src/main/java/com/prolificinteractive/materialcalendarview/CalendarDay.java]: No, let me clean that up.

File: library/src/main/java/com/prolificinteractive/materialcalendarview/CalendarDay.java (hunk 3 above)
Comment (quentin41500) [reply library/src/main/java/com/prolificinteractive/materialcalendarview/CalendarDay.java]: 👍

File: library/build.gradle (hunk 2 above)
Comment (quentin41500) [reply library/build.gradle]: Ok, we might have to if we remove `CalendarDay` later on however.

**Pull request merged**